surfactant/relationships/elf_relationship.py

```diff
@@ -42,107 +44,173 @@ def has_required_fields(metadata) -> bool:
-                continue
-
-            # check if the software entry is installed to one of the paths looked at for loading the dependency
-            for fp in fpaths:
-                if isinstance(item.installPath, Iterable) and fp in item.installPath:
-                    # software matching requirements to be the loaded dependency was found
-                    dependency_uuid = item.UUID
-                    rel = Relationship(dependent_uuid, dependency_uuid, "Uses")
-                    if rel not in relationships:
-                        relationships.append(rel)
+            fpaths = [p.joinpath(fname).as_posix() for p in default_search_paths]
+
+        # Phase 1: fs_tree lookup
+        matched_uuids = set()
+        used_method = {}
+
+        for path in fpaths:
+            match = sbom.get_software_by_path(path)
+            ok = bool(match and match.UUID != software.UUID)
+            logger.debug(f"[ELF][fs_tree] {path} → {'UUID=' + match.UUID if ok else 'no match'}")
+            if ok:
+                matched_uuids.add(match.UUID)
+                used_method[match.UUID] = "fs_tree"
+
+        # Phase 2: Legacy installPath fallback
+        if not matched_uuids:
+            for item in sbom.software:
+                has_name = isinstance(item.fileName, Iterable) and fname in (item.fileName or [])
+                if not has_name:
+                    continue
+                for fp in fpaths:
+                    if isinstance(item.installPath, Iterable) and fp in (item.installPath or []):
+                        if item.UUID != software.UUID:
+                            logger.debug(f"[ELF][legacy] {fname} in {fp} → UUID={item.UUID}")
+                            matched_uuids.add(item.UUID)
+                            used_method[item.UUID] = "legacy_installPath"
+
+        # Phase 3: Symlink-aware heuristic
+        # If path-based and legacy matching failed, fall back to checking:
+        # - Same fileName as dependency
+        # - Located in the same directory as any search path
+        if not matched_uuids and fname:
+            for item in sbom.software:
+                has_name = isinstance(item.fileName, Iterable) and fname in (item.fileName or [])
+                if not has_name or not isinstance(item.installPath, Iterable):
+                    continue
+                for ipath in item.installPath or []:
+                    ip_dir = pathlib.PurePosixPath(ipath).parent
+                    for fp in fpaths:
+                        if pathlib.PurePosixPath(fp).parent == ip_dir:
+                            if item.UUID != software.UUID:
+                                logger.debug(
+                                    f"[ELF][heuristic] {fname} via {ipath} ~ {fp} → UUID={item.UUID}"
+                                )
+                                matched_uuids.add(item.UUID)
+                                used_method[item.UUID] = "heuristic"
```

Comment:
Phase 3 is essentially doing the same thing as Phase 2 except without verifying that the file name at the end of the install path is correct, opening up the possibility of false positives. For a dependency of "somedir/abc", the full path must match, which includes the file name. With this heuristic, we end up with a scenario where a software entry like this would create a false positive:

file names = ["abc", "def"]
install paths = ["somedir/def", "anotherdir/abc"]

No match should be found because "abc" doesn't exist in "somedir" (perhaps an indication that something is broken on the system we got the filesystem dump from, but we aren't trying to fix a broken device).